### ISSUE TITLE
feat: Optimize reactive queries - tweak query results based change streams instead of re-running them (use EventReduce?)

### ISSUE THREAD

**Issue by beorn** [maxnowack/signaldb#1021]
Currently it seems that all queries are re-run when changes happen, which can be very slow both locally and result in additional queries to the remote server (in the case of AutoFetchCollection).

RxDB implements an Event Reduce algorithm that updates query results based on database change events, which is more optimal.   This will allow much more expensive and long persisted remote queries in AutoFetchCollection (since they don't have to be re-run all the time), and presumably more efficient reactivity updates internally too.

- https://rxdb.info/why-nosql.html#event-optimization
- https://github.com/pubkey/event-reduce

**Comment by maxnowack** [comment]
Good idea! PR appreciated!
For the `AutoFetchCollection` it wouldn't change that much, as queries only being registered at the first run and unregistered when the reactive scope is disposed. Reruns of queries shouldn't trigger a new registering/fetching from the server.

**Comment by maxnowack** [comment]
Closing this as stale. I probably will implement something similar with the next major version

**Comment by meerkapp** [comment]
It looks interesting. 🤔

**Comment by meerkapp** [comment]
@maxnowack is there something similar in v2 already?

**Comment by maxnowack** [comment]
@meerkapp no, I didn't implement something similar in v2. But this can be handled at DataAdapter level with a custom implementation. So it should be much easier to implement compared to v1 🙂
